test/helpers/user_helpers.rb

```diff
@@ -1,24 +1,23 @@
 module UserHelpers
-  def signin(user_or_options)
-    email = user_or_options.is_a?(Hash) ? user_or_options[:email] : user_or_options.email
-    password = user_or_options.is_a?(Hash) ? user_or_options[:password] : user_or_options.password
+  def signin_with(user = nil, email: nil, password: nil, second_step: true, set_up_2sv: true)
+    user ||= User.find_by(email: email)
+    email ||= user.email
+    password ||= user.password
+
+    if user && user.require_2sv && user.otp_secret_key.blank? && set_up_2sv
```

Comment:
Since it forms part of a boolean expression, it would be more idiomatic to express `require_2sv` in the predicate form.